src/backend/auxil/range_alloc.rs

```diff
@@ -0,0 +1,169 @@
+use std::ops::{Add, AddAssign, Range, Sub};
+use std::fmt::Debug;
+
+#[derive(Derivative)]
+#[derivative(Debug)]
+pub struct RangeAllocator<T> {
+    /// The range this allocator covers.
+    range: Range<T>,
```

Comment:
Maybe something like 'initial_range'  or ' space' to avoid repeating 'range' so often?